tests/selenium/pbtest.py

```diff
@@ -155,14 +147,26 @@ def run(self, result=None):
         else:
             manager = chrome_manager
 
-        with xvfb_manager(env) as xvfb:
-            with manager() as (driver, base_url):
-                self.base_url = base_url
-                self.xvfb = xvfb
-                self.driver = driver
-                self.js = self.driver.execute_script
-                super(PBSeleniumTest, self).run(result)
-
+        nretries = attempts.get(result.name, 1)
+        for i in range(nretries):
+            try:
+                with xvfb_manager(env) as xvfb:
+                    with manager() as (driver, base_url):
+
+                        self.base_url = base_url
+                        self.xvfb = xvfb
+                        self.driver = driver
+                        self.js = self.driver.execute_script
+
+                        super(PBSeleniumTest, self).run(result)
+                        if result._excinfo:
```

Comment:
This is a dirty hack to get a nice `@repeat_if_failed` decorator working. It only works with pytest. It should be removed once the tests are more consistent.